### PR TITLE
changefeedccl: misc fixes for tenant compatibility

### PR DESCRIPTION
**Pull request by dt**
see commits -- mostly just avoiding using NodeID, plus don't read meta2 directly.

### CODE REVIEW

File: pkg/ccl/changefeedccl/changefeed_dist.go [hunk 1]
```diff
@@ -107,18 +107,15 @@ func distChangefeedFlow(
...
 	var noTxn *kv.Txn
-	gatewayNodeID, err := execCfg.NodeID.OptionalNodeIDErr(48274)
-	if err != nil {
-		return err
-	}
+
 	dsp := execCtx.DistSQLPlanner()
 	evalCtx := execCtx.ExtendedEvalContext()
-	planCtx := dsp.NewPlanningCtx(ctx, evalCtx, nil /* planner */, noTxn, true /* distribute */)
+	planCtx := dsp.NewPlanningCtx(ctx, evalCtx, nil /* planner */, noTxn, execCfg.Codec.ForSystemTenant() /* distribute */)
```
Comment (HonoreDB) [inline pkg/ccl/changefeedccl/changefeed_dist.go:113]: This is temporary, right? Eventually tenants will be distributed?

**Review by HonoreDB — APPROVED** [review]

**Review by ajwerner — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r1, 2 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57805#-:-MOHsmx33TIyWXzZfrv6:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt and @HonoreDB)

___
*[pkg/ccl/changefeedccl/changefeed_dist.go, line 113 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/57805#-MODgbz6EjoDTU0ii1J_-r3-113:-MOHsAZQCcV2qvfXfYHc:b-2nlfw0) ([raw file](https://github.com/cockroachdb/cockroach/blob/382a2296c11bb59f3d907c840fde49b30b67e0b8/pkg/ccl/changefeedccl/changefeed_dist.go#L113)):*
<details><summary><i>Previously, HonoreDB (Aaron Zinger) wrote…</i></summary><blockquote>

This is temporary, right? Eventually tenants will be distributed?
</blockquote></details>

It's not in any immediate roadmap. The KV layer is distributed but for now the SQL execution is all local. 
___
*[pkg/ccl/changefeedccl/kvfeed/scanner.go, line 154 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/57805#-MOHsWvuAZMcguhho8f7:-MOHsWvv44WVyoruUQKk:b-qolcdx) ([raw file](https://github.com/cockroachdb/cockroach/blob/382a2296c11bb59f3d907c840fde49b30b67e0b8/pkg/ccl/changefeedccl/kvfeed/scanner.go#L154)):*
> ```
> 	ctx context.Context, ds *kvcoord.DistSender, targetSpans []roachpb.Span,
> ) ([]roachpb.Span, error) {
> 	ranges, err := allRangeSpans(ctx, ds, targetSpans)
> ```
nit: push the unwrapped down to here so that in the backport it's easier to just stick an `if` around getting the ranges using the new logic vs old logic.

**Review by dt — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57805)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner and @HonoreDB)

___
*[pkg/ccl/changefeedccl/kvfeed/scanner.go, line 154 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/57805#-MOHsWvuAZMcguhho8f7:-MQTIfvb1fr2Q9-I54ba:by2dmfr) ([raw file](https://github.com/cockroachdb/cockroach/blob/382a2296c11bb59f3d907c840fde49b30b67e0b8/pkg/ccl/changefeedccl/kvfeed/scanner.go#L154)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

> ```
> 	ctx context.Context, ds *kvcoord.DistSender, targetSpans []roachpb.Span,
> ) ([]roachpb.Span, error) {
> 	ranges, err := allRangeSpans(ctx, ds, targetSpans)
> ```
nit: push the unwrapped down to here so that in the backport it's easier to just stick an `if` around getting the ranges using the new logic vs old logic. 
</blockquote></details>

I think I'm fine backporting the whole change (or none of it), rather than having the backport do an extra if/else here.

**Review by dt — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57805#-:-MQXIGY58ojJmpvZTfwx:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner and @HonoreDB)

**Pull request merged**